src/directives/formly-field.js

```diff
@@ -59,18 +59,25 @@ function formlyField($http, $q, $compile, $templateCache, $interpolate, formlyCo
 
     // function definitions
     function runExpressions() {
+      const deferred = $q.defer()
```

Comment:
Not that my opinion matters much at this point, but you could actually clean up this code a little by returning the promise from `$timeout` and `$q.all`. That way you wouldn't need `$q.defer()`. So it'd look like:

```javascript
function runExpressions() {
  // must run on next tick to make sure that the current value is correct.
  return $timeout(function runExpressionsOnNextTick() {
    const promises = []
    const field = $scope.options
    const currentValue = valueGetterSetter()
    angular.forEach(field.expressionProperties, function runExpression(expression, prop) {
      const setter = $parse(prop).assign
      const promise = $q.when(formlyUtil.formlyEval($scope, expression, currentValue, currentValue))
        .then(function setFieldValue(value) {
          setter(field, value)
        })
      promises.push(promise)
    })
    return $q.all(promises)
  }, 0, false)
}
```